File: WinPaletter/GUI/Elements/Others/TreeView.vb
Imports System.ComponentModel

Namespace UI.WP

    <Description("TreeView fixed to respect dark/light mode")>
    Public Class TreeView : Inherits Windows.Forms.TreeView
        Protected Overrides ReadOnly Property CreateParams As CreateParams
            Get
                Dim parms As CreateParams = MyBase.CreateParams
                parms.Style = parms.Style Or &H80
                Return parms
            End Get
        End Property
    End Class

End Namespace

File: WinPaletter/GUI/Elements/Others/TreeView.cs
using System.ComponentModel;
using System.Windows.Forms;

namespace WinPaletter.UI.WP
{
    [Description("TreeView fixed to respect dark/light mode")]
    public class TreeView : System.Windows.Forms.TreeView
    {
        protected override CreateParams CreateParams
        {
            get
            {
                CreateParams parms = base.CreateParams;
                parms.Style |= 0x80;
                return parms;
            }
        }
    }
}